src/App/Components/CreateArticle/UpdateArticle.js

```diff
@@ -0,0 +1,385 @@
+/* eslint-disable no-shadow */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { Redirect } from 'react-router-dom';
+import propTypes from 'prop-types';
+import jwt from 'jwt-decode';
+
+import SimplexEditor from 'simple-react-editor';
+import Tagify from 'react-tagify-section';
+import queryString from 'query-string';
+import Navbar from '../Common/NavProfile/navbar';
+import Footer from '../Common/Footer';
+import Loader from '../Common/loader';
+import {
+  categories as getCategories,
+  getTags,
+  changeState,
+  drafting,
+  publish
+} from '../../../Redux/Actions/articles';
+import updateArticle from '../../../Redux/Actions/getArticle';
+import imageUploader from '../../../Helpers/image-upload';
+import Selector from './Selector';
+import NotFound from '../ArticleNotFound/ArticleNotFound';
+
+const { getArticleDetail, getArticleTags, getDraftedArticle } = updateArticle;
+export class UpdateArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      article: null,
+      categories: [],
+      tags: [],
+      imgUrl: null,
+      isCoverImage: false,
+      stateTags: [],
+      initialState: null,
+      published: false,
+      isLoading: true,
+      title: '',
+      description: '',
+      isChanged: false
+    };
+
+    /* istanbul ignore next */
+    this.autosaveFunctionality = setInterval(() => {
+      const {
+        stateTags, article, imgUrl, isChanged
+      } = this.state;
+      const { drafting: autosaveArticle, match } = this.props;
+      const { slug } = match.params;
+      const data = {
+        ...article, slug, tags: stateTags, coverImage: imgUrl
+      };
+      if (isChanged) {
+        autosaveArticle(data);
+        this.setState({ isChanged: false });
+      }
+    }, 6000);
+  }
+
+  componentWillMount() {
+    const {
```

Comment:
@truestbyheart same here.

Reply:
Same thing...doing the initial setup before the component mounts... I want to make sure when thecomponent mounts it has everything it requires to render properly

Reply:
@truestbyheart  Let's leave this but for future purposes, read about the React team's stance on `componentWillMount` here:
https://reactjs.org/docs/react-component.html

Reply:
Yaaah... I think i have to learn about react hooks... @MemunaHaruna